inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/RedisSource.java

```diff
@@ -17,54 +17,279 @@
+        this.snapShot = profile.get(TaskConstants.TASK_REDIS_OFFSET, "-1");
+        this.dbNumber = profile.get(TaskConstants.TASK_REDIS_DB_NUMBER, "0");
+        this.keys = new ConcurrentSkipListSet<>(Arrays.asList(profile.get(TaskConstants.TASK_REDIS_KEYS).split(",")));
+        this.isSubscribe = profile.getBoolean(TaskConstants.TASK_REDIS_IS_SUBSCRIBE, false);
+        this.instanceId = profile.getInstanceId();
+        this.redisQueue = new LinkedBlockingQueue<>(REDIS_QUEUE_SIZE);
+        initGson();
+        String uri = getRedisUri();
+        try {
+            if (isSubscribe) {
+                // use subscribe mode
+                this.executor = (ScheduledExecutorService) Executors.newSingleThreadExecutor();
+                redisReplicator = new RedisReplicator(uri);
+                initReplicator();
+                executor.execute(startReplicatorSync());
+            } else {
+                this.executor = Executors.newScheduledThreadPool(1);
+                // use command mode
+                this.redisCommand = profile.get(TaskConstants.TASK_REDIS_COMMAND, "get");
+                this.fieldOrMember = profile.get(TaskConstants.TASK_REDIS_FIELD_OR_MEMBER, null);
+                // default frequency 1min
+                long syncFreq = profile.getLong(TaskConstants.TASK_REDIS_SYNC_FREQ, 60 * 1000);
+                this.jedis = new Jedis(uri);
+                jedis.connect();
+                executor.scheduleWithFixedDelay(startJedisSync(), 0, syncFreq, TimeUnit.MILLISECONDS);
+            }
+        } catch (URISyntaxException | IOException | JedisConnectionException e) {
+            sourceMetric.pluginReadFailCount.addAndGet(1);
+            LOGGER.error("Connect to redis {}:{} failed.", hostName, port, e);
+        }
+    }
 
+    private Runnable startReplicatorSync() {
+        return () -> {
+            AgentThreadFactory.nameThread(getThreadName() + "redis subscribe mode");
+            executor.execute(new Thread(() -> {
+                try {
+                    this.redisReplicator.open();
+                } catch (IOException e) {
+                    LOGGER.error("Redis source error, fail to start replicator", e);
+                }
+            }));
+        };
     }
 
-    @Override
-    protected void printCurrentState() {
+    private Runnable startJedisSync() {
+        return () -> {
+            AgentThreadFactory.nameThread(getThreadName() + "redis command mode");
+            executor.execute(new Thread(() -> {
+                Map<String, Object> dataMap =
+                        fetchDataByJedis(jedis, redisCommand, new ArrayList<>(keys), fieldOrMember);
+                synchronizeData(gson.toJson(dataMap));
+            }));
+        };
+    }
+
+    private Map<String, Object> fetchDataByJedis(Jedis jedis, String command, List<String> keys, String fieldOrMember) {
+        Map<String, Object> result = new HashMap<>();
+        switch (command.toUpperCase()) {
+            case "GET":
```

Comment:
Suggest changing the command string to a constant. I see that the same command is used in several places

Reply:
done, now i change the command string to a constant and use reflections to handle